WordPress/Classes/Networking/TaxonomyServiceRemoteREST.m

```diff
@@ -2,76 +2,148 @@
+                         failure:(void (^)(NSError *error))failure
+{
+    [self getTaxonomyWithType:TaxonomyServiceRemoteRESTCategoryTypeIdentifier
+                   parameters:[self parametersForPaging:paging]
+                      success:^(id responseObject) {
+                          if (success) {
+                              success([self remoteCategoriesWithJSONArray:[responseObject arrayForKey:@"categories"]]);
+                          }
+                      } failure:failure];
+}
+
+- (void)searchCategoriesWithName:(NSString *)nameQuery
+                   success:(void (^)(NSArray <RemotePostCategory *> *tags))success
+                   failure:(void (^)(NSError *error))failure
+{
+    NSParameterAssert(nameQuery.length > 0);
+    [self getTaxonomyWithType:TaxonomyServiceRemoteRESTCategoryTypeIdentifier
+                   parameters:@{@"search": nameQuery}
+                      success:^(id responseObject) {
+                          if (success) {
+                              success([self remoteCategoriesWithJSONArray:[responseObject arrayForKey:@"categories"]]);
+                          }
+                      } failure:failure];
+}
+
+#pragma mark - tags
+
+- (void)getTagsWithSuccess:(void (^)(NSArray <RemotePostTag *> *tags))success
+                   failure:(void (^)(NSError *error))failure
+{
+    [self getTagsWithSuccess:success
+                      paging:nil
+                     failure:failure];
+}
+
+- (void)getTagsWithSuccess:(void (^)(NSArray<RemotePostTag *> *))success
+                    paging:(RemoteTaxonomyPaging *)paging
+                   failure:(void (^)(NSError *))failure
+{
+    [self getTaxonomyWithType:TaxonomyServiceRemoteRESTTagTypeIdentifier
+                   parameters:[self parametersForPaging:paging]
+                      success:^(id responseObject) {
+                          if (success) {
+                              success([self remoteTagsWithJSONArray:[responseObject arrayForKey:@"tags"]]);
+                          }
+                      } failure:failure];
+}
+
+- (void)searchTagsWithName:(NSString *)nameQuery
+                   success:(void (^)(NSArray <RemotePostTag *> *tags))success
+                   failure:(void (^)(NSError *error))failure
+{
+    NSParameterAssert(nameQuery.length > 0);
+    [self getTaxonomyWithType:TaxonomyServiceRemoteRESTTagTypeIdentifier
+                   parameters:@{@"search": nameQuery}
+                      success:^(id responseObject) {
+                          if (success) {
+                              success([self remoteTagsWithJSONArray:[responseObject arrayForKey:@"tags"]]);
+                          }
+                      } failure:failure];
+}
+
+#pragma mark - default methods
+
+- (void)createTaxonomyWithType:(NSString *)typeIdentifier
+                    parameters:(NSDictionary *)parameters
+                       success:(void (^)(id responseObject))success
```

Comment:
Looks like `responseObject` is always going to be an `NSDictionary`?  Maybe just declare it that way instead of `id`.